### PR TITLE
Make the egherkin work as a dependency in other rebar3 projects

### PR DESCRIPTION
**Pull request by andreashasse**
Hello,

I've been playing around with egherkin. To be able to use it as a dependency in a rebar3 project, I had to make some small modifications. On top of that, I added a GitHub workflow and some minor fixes. The commits should be somewhat self-contained if you want me to exclude one of them.

### CODE REVIEW

File: src/egherkin.app.src [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{application, egherkin,
```
Comment (eboutin) [inline src/egherkin.app.src:1]: Is it possible to have ebin/egherkin.app generated by rebar3 without this? With erlang.mk, this file is optional and erlang.mk automates the generation of the .app file from all the info available in the makefile

**Review by eboutin — COMMENTED** [review]
Thank you very much for this clean commit. My only issue is with the .app.src file which I usually avoid since erlang.mk does an excellent job at generating the .app file from the makefile parameters. Tell me if rebar3 cannot do the same and I'll approve the PR.

File: src/egherkin.app.src (hunk 1 above)
Comment (andreashasse) [reply src/egherkin.app.src:1]: I'll have a look.

**Comment by andreashasse** [comment]
I added support for linux newlines.

File: src/egherkin.app.src (hunk 1 above)
Comment (eboutin) [reply src/egherkin.app.src:1]: I checked rebar3's documentation. My understanding is that the .app.src file is needed by rebar to generate the .app file.

**Pull request merged**